src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -202,22 +255,118 @@ func CreateAdmin(
 	fs afero.Fs,
 	brokers func() []string,
 	configuration func() (*config.Config, error),
+	tlsConfig func() (*config.TLS, error),
+	authConfig func() (*config.SCRAM, error),
 ) func() (sarama.ClusterAdmin, error) {
 	return func() (sarama.ClusterAdmin, error) {
+		var err error
 		conf, err := configuration()
 		if err != nil {
 			return nil, err
 		}
-		cfg, err := kafka.LoadConfig(conf)
+
+		tls, err := tlsConfig()
+		if err != nil {
+			return nil, err
+		}
+		// If no TLS config was set, try to look for TLS config in the
+		// config file.
+		if tls == nil {
+			tls = &conf.Rpk.TLS
+		}
+
+		scram, err := authConfig()
+		if err != nil {
+			return nil, err
+		}
+		// If no SCRAM config was set, try to look for it in the
+		// config file.
+		if scram == nil {
+			scram = &conf.Rpk.SCRAM
+		}
+
+		cfg, err := kafka.LoadConfig(tls, scram)
 		if err != nil {
 			return nil, err
 		}
+
 		bs := brokers()
 		admin, err := sarama.NewClusterAdmin(bs, cfg)
 		return admin, wrapConnErr(err, bs)
 	}
 }
 
+func KafkaAuthConfig(
+	user, password, mechanism *string,
+) func() (*config.SCRAM, error) {
+	return func() (*config.SCRAM, error) {
+		if *user == "" && *password == "" {
+			return nil, nil
+		}
+		if *user == "" && *password != "" {
+			return nil, errors.New("empty user. Pass --user to set a value.")
+		}
+		if *user != "" && *password == "" {
+			return nil, errors.New("empty password. Pass --password to set a value.")
+		}
+		if *mechanism != sarama.SASLTypeSCRAMSHA256 && *mechanism != sarama.SASLTypeSCRAMSHA512 {
+			return nil, fmt.Errorf(
+				"unsupported mechanism '%s'. Pass --%s to set a value."+
```

Comment:
NIT: Can the error provide more context to end user? 
```suggestion
				"The user and password is provided. The mechanism '%s' is unsupported. Pass --%s to set a value."+
```

Reply:
That looks a little confusing to me. I think the error should point out exactly what needs to be fixed, which is that the mechanism is unsupported (it also tells the user which mechanisms *are* supported.)